modules/sdk/src/main/scala/org/tessellation/sdk/infrastructure/snapshot/storage/SnapshotStorage.scala

```diff
@@ -124,14 +124,22 @@ object SnapshotStorage {
         hashCache(hash).set(snapshot.some) >>
           ordinalCache(snapshot.ordinal).set(hash.some) >>
           snapshotLocalFileSystemStorage.write(snapshot).handleErrorWith { e =>
-            logger.error(e)(s"Failed writing snapshot to disk! hash=$hash ordinal=${snapshot.ordinal}") >>
-              notPersistedCache.update(current => current + snapshot.ordinal)
+            snapshotExists(snapshot).ifM(
+              logger.info(s"Snapshot is already saved on disk. hash=$hash ordinal=${snapshot.ordinal}"),
+              logger.error(e)(s"Failed writing snapshot to disk! hash=$hash ordinal=${snapshot.ordinal}") >>
+                notPersistedCache.update(current => current + snapshot.ordinal)
+            )
```

Comment:
It checks whether snapshot exists under the hash, but it should exist under the ordinal as well, so either update `snapshotExists` to check both hash and ordinal, or update `notPersistedCache` as it was before.

Reply:
bump @kriscieslak

Reply:
I have improved snapshotExists method to check by hash and ordinal name.